core/spec/models/spree/tax/taxation_integration_spec.rb

```diff
@@ -0,0 +1,801 @@
+    create :product,
+           price: 10,
+           name: "Download",
+           tax_category: digital_category,
+           shipping_category: digital_shipping_category
+  end
+  let(:sweater_product) do
+    create :product,
+           price: 30,
+           name: "Download",
+           tax_category: normal_category,
+           shipping_category: normal_shipping_category
+  end
+
+  let(:book) { book_product.master }
+  let(:download) { download_product.master }
+  let(:sweater) { sweater_product.master }
+
+  let(:books_category) { create :tax_category, name: "Books" }
+  let(:normal_category) { create :tax_category, name: "Normal" }
+  let(:digital_category) { create :tax_category, name: "Digital Goods" }
+
+  let(:books_shipping_category) { create :shipping_category, name: "Book Shipping" }
+  let(:normal_shipping_category) { create :shipping_category, name: "Normal Shipping" }
+  let(:digital_shipping_category) { create :shipping_category, name: "Digital Premium Download" }
+
+  let(:line_item) { order.line_items.first }
+  let(:shipment) { order.shipments.first }
+  let(:shipping_rate) { shipment.shipping_rates.first }
+
+  context 'selling from germany' do
+    let(:germany) { create :country, iso: "DE" }
+    # The weird default_tax boolean is what makes this context one with default included taxes
+    let!(:germany_zone) { create :zone, countries: [germany], default_tax: true }
+    let(:romania) { create(:country, iso: "RO") }
+    let(:romania_zone) { create(:zone, countries: [romania] ) }
+    let(:eu_zone) { create(:zone, countries: [romania, germany]) }
+    let(:world_zone) { create(:zone, :with_country) }
+
+    let!(:german_book_vat) do
+      create(
+        :tax_rate,
+        name: "German reduced VAT",
+        included_in_price: true,
+        amount: 0.07,
+        tax_category: books_category,
+        zone: eu_zone
+      )
+    end
+    let!(:german_normal_vat) do
+      create(
+        :tax_rate,
+        name: "German VAT",
+        included_in_price: true,
+        amount: 0.19,
+        tax_category: normal_category,
+        zone: eu_zone
+      )
+    end
+    let!(:german_digital_vat) do
+      create(
+        :tax_rate,
+        name: "German VAT",
+        included_in_price: true,
+        amount: 0.19,
+        tax_category: digital_category,
+        zone: germany_zone
+      )
+    end
+    let!(:romanian_digital_vat) do
+      create(
+        :tax_rate,
+        name: "Romanian VAT",
+        included_in_price: true,
+        amount: 0.24,
+        tax_category: digital_category,
+        zone: romania_zone
+      )
+    end
+    let!(:book_shipping_method) do
+      create :shipping_method,
+             cost: 8.00,
+             shipping_categories: [books_shipping_category],
+             tax_category: books_category,
+             zones: [eu_zone, world_zone]
+    end
+
+    let!(:sweater_shipping_method) do
+      create :shipping_method,
+             cost: 16.00,
```

Comment:
That would be more realistic, but the system under test here is the
taxation system. I'm assuming the shipping system works - the additional
benefit is that the rates used are always recognizable: 16 for sweaters, 8
for books and 2 for premium downloads ;)
Am 05.02.2016 19:32 schrieb "Thomas von Deyen" <notifications@github.com>:

> In core/spec/models/spree/tax/taxation_integration_spec.rb
> <https://github.com/solidusio/solidus/pull/799#discussion_r52053847>:
>
> > +        amount: 0.24,
> > +        tax_category: digital_category,
> > +        zone: romania_zone
> > +      )
> > +    end
> > +    let!(:book_shipping_method) do
> > +      create :shipping_method,
> > +             cost: 8.00,
> > +             shipping_categories: [books_shipping_category],
> > +             tax_category: books_category,
> > +             zones: [eu_zone, world_zone]
> > +    end
> > +
> > +    let!(:sweater_shipping_method) do
> > +      create :shipping_method,
> > +             cost: 16.00,
>
> Is this a real world use case? Shouldn't we test different rates for each
> zone? I.e.: 6€ in German zone, 12€ in EU zone, 20€ for world?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/solidusio/solidus/pull/799/files#r52053847>.
>